LayoutTests/imported/w3c/web-platform-tests/import-maps/nonimport-integrity.html

```diff
@@ -0,0 +1,161 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+let log = [];
+</script>
+<script type="importmap">
+{
+  "integrity": {
+    "./resources/log.js?pipe=sub&name=ModuleNoIntegrity": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=ModuleIntegrity": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=ModuleEmptyIntegrity": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=ModuleBadIntegrityAttribute": "sha384-COhDkp+ybIZ9wz9hUaSJ5NzKcn8wOMZMpsACZfTeEdBRtNcX5yWJnFn+lIK77Tay",
+    "./resources/log.js?pipe=sub&name=ModulePreloadNoIntegrity": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=ModulePreloadIntegrity": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=ModulePreloadEmptyIntegrity": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=ModulePreloadBadIntegrityAttribute": "sha384-026dlUs9+KSmPb0Uc7oUPOlWBO67o7vSFdfLJZWEVTvKCly5NXO8+CsOXl54ZBqJ",
+    "./resources/log.js?pipe=sub&name=NonModule": "sha384-foobar",
+    "/images/green.png": "sha384-foobar"
+  }
+}
+</script>
+<script type="module">
+promise_test(async t => {
+  log = [];
+  const script = document.createElement("script");
+  script.type = "module";
+  script.src = "./resources/log.js?pipe=sub&name=ModuleNoIntegrity";
+  const promise = new Promise((resolve, reject) => {
+    script.onload = resolve;
+    script.onerror = () => { reject(Error()); };
+  });
+  document.head.appendChild(script);
+  await promise_rejects_js(t, Error, promise);
```

Comment:
Maybe consider doing it this way? 

```suggestion
  const promise = new Promise((resolve, reject) => {
    script.onload = reject(new Error("It shouldn't have loaded"));
    script.onerror = resolve;
  });
  document.head.appendChild(script);
  const event = await promise;
  assert_equals(event.type, "error");
```

Reply:
Done!